pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -312,12 +319,22 @@ func (s *FlinkStateMachine) handleApplicationSavepointing(ctx context.Context, a
 
 	// check the savepoints in progress
 	savepointStatusResponse, err := s.flinkController.GetSavepointStatus(ctx, application, application.Status.DeployHash)
-	if err != nil {
+	// Here shouldRollback() is used as a proxy to identify if we have exhausted all retries trying to GetSavepointStatus
+	// The application is NOT actually rolled back  because we're in a spot where there may be no application running and we don't
+	// have information on the last successful savepoint.
+	// In the event that rollback evaluates to True, we don't return immediately but allow for the remaining steps to proceed as normal.
```

Comment:
I think that if we fail to get the savepoint status, we will still not be able to recover. The following steps (checking for externalized checkpoint etc.) still require that we have a non-nil savepointStatusResponse.

Reply:
True. This was a bad assumption (that we get return the response as is with the error), reinforced by bad unit tests. Fixed both, thanks for pointing out.